File: src/main/java/com/rs/java/game/area/banks/FishingGuildBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class FishingGuildBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Fishing Guild Bank";
	}

	public FishingGuildBank() {
		super(
				new Polygon(new WorldTile[] {
				new WorldTile(2583, 3420, 0),
				new WorldTile(2583, 3424, 0),
				new WorldTile(2587, 3424, 0),
				new WorldTile(2587, 3420, 0),
				new WorldTile(2583, 3420, 0),
		}) );
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/FishingGuildBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class FishingGuildBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2583, 3420, 0),
            WorldTile(2583, 3424, 0),
            WorldTile(2587, 3424, 0),
            WorldTile(2587, 3420, 0),
            WorldTile(2583, 3420, 0),
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Fishing Guild Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
